pandas/core/frame.py

```diff
@@ -1336,7 +1343,12 @@ def to_numpy(self, dtype=None, copy: bool = False) -> np.ndarray:
         array([[1, 3.0, Timestamp('2000-01-01 00:00:00')],
                [2, 4.5, Timestamp('2000-01-02 00:00:00')]], dtype=object)
         """
-        result = np.array(self.values, dtype=dtype, copy=copy)
+        arr = [
+            self[c].to_numpy(dtype=dtype, na_value=na_value, **kwargs)
+            for c in self.columns
+        ]
+        result = np.array(arr).T
```

Comment:
pass thru copy=, though I think this may actually *always* copy

Reply:
we should add a test for an all float frame to see if it actually does

Reply:
we *could* a special case for a single dtype (maybe)

Reply:
That's tested here and indeed copy=False doesn't do anything after this change:
https://github.com/pandas-dev/pandas/blob/085752f95adb625a43aef64481be56a393e8732a/pandas/tests/frame/test_api.py#L363
I think this is okay since now the copy behavior isn't guaranteed anyways when copy=False (would want to explicitly state this in the docstring probably)?

Reply:
We should preserve the no-copy behaviour for a single block, I would say (so the existing test keeps working). 

@dsaxton we should maybe only do the column-wise approach as you do above when there is at least one extension dtype column (then we are sure the dataframe is not a single 2D block)